dbt/adapters/snowflake/impl.py

```diff
@@ -263,6 +263,15 @@ def list_relations_without_caching(
         # this can be collapsed once Snowflake adds is_iceberg to show objects
         columns = ["database_name", "schema_name", "name", "kind", "is_dynamic"]
         if self.behavior.enable_iceberg_materializations.no_warn:
+            # The QUOTED_IDENTIFIERS_IGNORE_CASE setting impacts column names like
+            # is_iceberg which is created by dbt, but it does not affect the case
+            # of column values in Snowflake's SHOW OBJECTS query! This normalization
+            # step ensures consistent scanning of the is_iceberg column as a
+            # lowercase name, treating it as if it were a system-provided field.
+            #
+            # This along with the behavior flag in general may be removed when
+            # is_iceberg comes on show objects.
+            schema_objects = schema_objects.rename(column_names={"IS_ICEBERG": "is_iceberg"})
```

Comment:
As a sanity check, did you confirm that the new test below fails when this line is commented out?

Reply:
Yep I wrote this in the description of the PR. I explicitly reproduced, got the fail on the test, then added this and watched it go green.

Reply:
Oh one added wrinkle is I temporarily set the QUOTED_IDENTIFIERS_IGNORE_CASE on in the macro. I didn't capture that in the test. Let me rework for that. Glad you said something as this test is missing a configuration!

Reply:
With the new test I have properly verified removing this line reproduces the error in the test itself.